Building with Bloom module fails due to undefined `SHA256_xxx` functions

Recently one of the cloud builds failed with this error:

```
modules/.output/eagle/debug/lib/libmodules.a(bloom.o): In function `bloom_filter_reset':
/home/travis/build/marcelstoer/nodemcu-custom-build/nodemcu-firmware/app/modules/bloom.c:125: undefined reference to `SHA256_Init'
/home/travis/build/marcelstoer/nodemcu-custom-build/nodemcu-firmware/app/modules/bloom.c:125: undefined reference to `SHA256_Update'
/home/travis/build/marcelstoer/nodemcu-custom-build/nodemcu-firmware/app/modules/bloom.c:87: undefined reference to `SHA256_Final'
modules/.output/eagle/debug/lib/libmodules.a(bloom.o): In function `add_or_check':
/home/travis/build/marcelstoer/nodemcu-custom-build/nodemcu-firmware/app/modules/bloom.c:23: undefined reference to `SHA256_Init'
/home/travis/build/marcelstoer/nodemcu-custom-build/nodemcu-firmware/app/modules/bloom.c:26: undefined reference to `SHA256_Update'
/home/travis/build/marcelstoer/nodemcu-custom-build/nodemcu-firmware/app/modules/bloom.c:29: undefined reference to `SHA256_Final'
collect2: error: ld returned 1 exit status
```

I see that [`bloom.c` requires `SHA2_ENABLE` to be defined](https://github.com/nodemcu/nodemcu-firmware/blob/master/app/modules/bloom.c). That is fine; it's on by default and I'm not getting that 

> Must have SHA2_ENABLE set for BLOOM module

message. Yet, it appears the functions defined in `app/crypto/sha2.h` are not found.

What is going wrong here?

--- a/app/Makefile
+++ b/app/Makefile
@@ -20,7 +20,7 @@ FLAVOR = debug
 ifndef PDIR # {
 GEN_IMAGES= eagle.app.v6.out
 GEN_BINS= eagle.app.v6.bin
-OPT_MKTARGETS        := coap crypto dht http mqtt pcm sjson sqlite3 tsl2561 websocket
+OPT_MKTARGETS        := coap dht http mqtt pcm sjson sqlite3 tsl2561 websocket
 OPT_MKLIBTARGETS     := u8g2 ucg
 SEL_MKTARGETS        := $(shell $(CC) -E -dM include/user_modules.h | sed -n '/^\#define LUA_USE_MODULES_/{s/.\{24\}\(.*\)/\L\1/; p}')
 OPT_SEL_MKLIBTARGETS := $(foreach tgt,$(OPT_MKLIBTARGETS),$(findstring $(tgt), $(SEL_MKTARGETS)))
@@ -31,6 +31,7 @@ SPECIAL_MKTARGETS    :=$(APP_MKTARGETS)
 
 SUBDIRS= 					\
 	user 					\
+	crypto					\
 	driver 					\
 	mbedtls 				\
 	platform 				\
@@ -63,6 +64,7 @@ LD_FILE = $(LDDIR)/nodemcu.ld
 
 COMPONENTS_eagle.app.v6 = 			\
 	user/libuser.a 				\
+	crypto/libcrypto.a			\
 	driver/libdriver.a 			\
 	platform/libplatform.a 			\
 	task/libtask.a 				\

